app/models/provider.rb

```diff
@@ -48,6 +48,14 @@ def accredited_courses
     Course.where(accredited_provider: self)
   end
 
+  def acts_as_accredited_body_in_current_cycle?
+    accredited_courses.current_cycle.exposed_in_find.any?
+  end
+
+  def all_courses_open_in_current_cycle?
+    (accredited_courses.current_cycle.exposed_in_find + courses.current_cycle.exposed_in_find).all?(&:open_on_apply?)
```

Comment:
```suggestion
    (accredited_courses.or(courses)).current_cycle.exposed_in_find.all?(&:open_on_apply?)
```
Is a bit more concise and seems to do the same thing

Reply:
Tidy, thank you!